user-service/src/main/java/com/local_dating/user_service/infrastructure/repository/UserRecomCardRepositoryCustom.java

```diff
@@ -0,0 +1,78 @@
+package com.local_dating.user_service.infrastructure.repository;
+
+import com.local_dating.user_service.domain.entity.QUserPreference;
+import com.local_dating.user_service.domain.entity.QUserRecomCard;
+import com.local_dating.user_service.domain.entity.UserPreference;
+import com.local_dating.user_service.domain.entity.UserRecomCard;
+import com.local_dating.user_service.domain.vo.UserPreferenceCountVO2;
+import com.querydsl.core.types.Projections;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Repository
+@RequiredArgsConstructor
+public class UserRecomCardRepositoryCustom {
+
+    private final JPAQueryFactory queryFactory;
+
+    public List<UserRecomCard> findValidCard(String userId) {
+
+        QUserRecomCard userRecomCard = QUserRecomCard.userRecomCard;
+
+        //LocalDateTime now = LocalDateTime.now();
+        //LocalDateTime endDate = now.minusDays(14);
+        LocalDateTime endDate = LocalDateTime.now().minusDays(14);
+
+        return queryFactory
+                .select(userRecomCard)
+                .from(userRecomCard)
+                .where(userRecomCard.userId.eq(userId)
+                                .and(userRecomCard.useyn.eq("Y"))
+                                .and(userRecomCard.inDate.gt(endDate))
```

Comment:
.and(userRecomCard.inDate.gt(Expressions.dateTemplate(LocalDateTime.class, "DATE_ADD({0}, INTERVAL 14 DAY)", userRecomCard.inDate))